### PR TITLE
Add more type annotations and fix warnings in Python code

### PR DESCRIPTION
**Pull request by cyyever**
### Description

Remove some 'type: ignore' and other fixes
### Motivation and Context
For better code.

### CODE REVIEW

File: onnx/test/test_backend_test.py [hunk 1]
```diff
@@ -68,12 +66,10 @@ def prepare(
     def run_node(
         cls,
         node: NodeProto,
-        inputs: Any,
-        device: str = "CPU",
-        outputs_info: Sequence[tuple[numpy.dtype, tuple[int, ...]]] | None = None,
-        **kwargs: Any,  # noqa: ARG003
-    ) -> tuple[Any, ...] | None:
-        super().run_node(node, inputs, device=device, outputs_info=outputs_info)
+        *args: Any,  # noqa: ARG003
```
Comment (xadupre) [inline onnx/test/test_backend_test.py:69]: I wonder why it was needed to change this piece of code.

File: onnx/backend/base.py [hunk 2]
```diff
@@ -83,11 +81,9 @@ def is_compatible(
 
     @classmethod
     def prepare(
-        cls, model: ModelProto, device: str = "CPU", **kwargs: Any  # noqa: ARG003
-    ) -> BackendRep | None:
-        # TODO Remove Optional from return type
+        cls, model: ModelProto, *args: Any, **kwargs: Any  # noqa: ARG003
```
Comment (xadupre) [inline onnx/backend/base.py:84]: I don't see the benefit in doing that. We lose some information. device is not used but I tihnk it is easier to keep it if anybody wants to use this piece of code and create his own backend.

Comment (cyyever) [reply onnx/backend/base.py:84]: Leaving unused parameters causes confusion. The demonstrations purpose should be assigned to examples.

File: onnx/test/test_backend_test.py (hunk 1 above)
Comment (cyyever) [reply onnx/test/test_backend_test.py]: reverted

**Review by justinchuby — APPROVED** [review]

**Pull request merged**